draft/0006-pat.md

```diff
@@ -0,0 +1,127 @@
+---
+Start date: 2024-08-15
+Author(s): wangsijie
+Revision: 1
+---
+
+# Personal Access Token (PAT)
+
+## Overview
+
+This RFC proposes an extension to Logto's authentication system to support Personal Access Tokens (PATs). This feature aims to provide users with a secure and convenient way to authenticate automated processes and integrations. PATs are long-lived tokens that can be used by non-interactive clients to exchange for short-lived access tokens through token exchange endpoints first introduced in RFC-0005 impersonation.
+
+## Motivation
+
+The current authentication flow in Logto is optimized for interactive user sessions. However, there are scenarios where users need to authenticate non-interactive processes or third-party integrations. These include:
+
+1. Automated scripts or bots that perform actions on behalf of the user.
+2. Integration with external tools and services that require API access.
+3. Long-running processes that outlast typical session durations.
+
+## Personal Access Token (PAT) Creation
+
+The client makes a post request to the tenant's Management API `/api/users/:userId/personal-access-tokens` to create a PAT for the user. The following parameters are included in the HTTP request entity-body using the `application/json` format.
+
+1. `name`: REQUIRED. A user-defined name for the PAT to identify its purpose, should be unique in the user's account.
+2. `expiresAt`: OPTIONAL. The epoch time in milliseconds when the token will expire. If not provided, the token will never expire.
+
+If the request to create a PAT is successful, the tenant's Management API returns the PAT in the response.
+
+1. `name`: REQUIRED. The name of the PAT.
+2. `value`: REQUIRED. The value of the PAT.
+3. `expiresAt`: OPTIONAL. The epoch time in milliseconds when the token will expire.
+
+### Example PAT creation
+
+```http
+POST /api/users/123/personal-access-tokens HTTP/1.1
+Host: tenant.logto.app
+Content-Type: application/json
+Authorization: Bearer <access-token-with-management-api-access>
+
+{
+  "name": "My PAT",
+}
+```
+
+```http
+HTTP/1.1 201 Created
+Content-Type: application/json
+
+{
+  "name": "My PAT",
+  "value": "pat_W51arOqe7nynW75nWhvYogyc",
+  "expiresAt": null
+}
+```
+
+## Token Exchange with PAT
```

Comment:
we should align the cases of "Token Exchange". e.g. line 11 and 60 are using "token exchange"

Reply:
Aligned, `Token exchange with PAT`.